renku_notebooks/util/kubernetes_.py

```diff
@@ -212,3 +213,63 @@ def read_namespaced_pod_log(pod_name, max_log_lines=0):
             pod_name, kubernetes_namespace, tail_lines=max_log_lines
         )
     return logs
+
+
+def create_or_replace_registry_secret(user, namespace):
+    """Read or replace a registry secret for a user."""
+    import base64
+    import json
+
+    token = _get_oauth_token(user)
+    payload = {
+        "auths": {
+            current_app.config.get("IMAGE_REGISTRY"): {
+                "Username": "oauth2",
+                "Password": token,
+                "Email": user.get("email"),
+            }
+        }
+    }
+
+    data = {
+        ".dockerconfigjson": base64.b64encode(json.dumps(payload).encode()).decode()
+    }
+
+    secret = client.V1Secret(
+        api_version="v1",
+        data=data,
+        kind="Secret",
+        metadata={
+            "name": f"{user.get('name')}-registry",
+            "namespace": kubernetes_namespace,
+            "annotations": {"renku.io/username": user.get("name")},
+            "labels": {
+                "component": "singleuser-server",
+                "renku.io/username": user.get("name"),
+            },
+        },
+        type="kubernetes.io/dockerconfigjson",
+    )
+
+    existing = _check_existing_secret(user, kubernetes_namespace)
```

Comment:
You could just use `v1.read_namespaced_secret(namespace, name)` which corresponds to `GET /api/v1/namespaces/{namespace}/secrets/{name}`.

Reply:
haha yep I could just do that - I massively over-complicated this. Not sure why.